client/lib/titan/new-mailbox.js

```diff
@@ -0,0 +1,245 @@
+		password: getMailboxRequiredStringValue(),
+	} );
+
+const sanitizeEmailSuggestion = ( emailSuggestion ) =>
+	String( emailSuggestion )
+		.replace( /[^0-9a-z_'.-]/gi, '' )
+		.toLowerCase();
+
+const validateMailboxesAreUnique = ( mailboxes ) => {
+	const mailboxNameCounts = mailboxes.reduce( ( nameCount, mailbox ) => {
+		nameCount[ mailbox.mailbox ] = 1 + nameCount[ mailbox.mailbox ] ?? 0;
+		return nameCount;
+	}, {} );
+
+	return mailboxes.map( ( mailbox ) => {
+		if ( mailboxNameCounts[ mailbox.mailbox ] > 1 ) {
+			return {
+				...mailbox,
+				[ mailbox ]: { value: mailbox.mailbox, error: translate( 'Please use unique mailboxes' ) },
+			};
+		}
+		return mailbox;
+	} );
+};
+
+const mapMailboxFields = ( mailbox, callback, skipFields = [] ) => {
+	return Object.entries( mailbox )
+		.map( ( entry ) => {
+			const fieldName = entry[ 0 ];
+			if ( fieldName === 'uuid' || skipFields.includes( fieldName ) ) {
+				return entry;
+			}
+			const fieldValue = entry[ 1 ];
+			return [ fieldName, callback( fieldValue, fieldName ) ];
+		} )
+		.reduce( ( updatedMailbox, entry ) => {
+			updatedMailbox[ entry[ 0 ] ] = entry[ 1 ];
+			return updatedMailbox;
+		}, {} );
+};
+
+const removePreviousError = ( { value } ) => ( { value, error: null } );
+
+const clearPreviousErrors = ( mailboxes ) => {
+	return mailboxes.map( ( mailbox ) => mapMailboxFields( mailbox, removePreviousError ) );
+};
+
+const validateRequiredField = ( { value, error } ) => ( {
+	value,
+	error:
+		! error && typeof value !== 'boolean' && ( ! value || '' === value.trim() )
+			? translate( 'This field is required.' )
+			: error,
+} );
+
+const validateRequiredMailboxFields = ( mailbox, optionalFields = [] ) => {
+	return mapMailboxFields( mailbox, validateRequiredField, optionalFields );
+};
+
+const validateFullEmailAddress = ( { value, error }, allowEmpty = false ) => {
+	if ( error ) {
+		return { value, error };
+	}
+	if ( allowEmpty && value === '' ) {
+		return { value, error };
+	}
+	const invalidEmailAddressError = translate( 'Please supply a valid email address.' );
+	const emailParts = value.split( '@' );
+	if ( emailParts.length !== 2 ) {
+		return {
+			value,
+			error: invalidEmailAddressError,
+		};
+	}
+
+	// TODO: improve this check to support internationalized SLDs and TLDs
+	const emailDomain = emailParts[ 1 ];
+	if ( ! /^[a-z0-9].*\.[a-z]{2,3}$/i.test( emailDomain ) ) {
+		return {
+			value,
+			error: invalidEmailAddressError,
+		};
+	}
+
+	const mailboxName = emailParts[ 0 ];
+	// Note that this expression allows + symbols in local names
+	if ( ! /^[0-9a-z_-](\.?[0-9a-z_+-])*$/i.test( mailboxName ) ) {
+		return {
+			value,
+			error: invalidEmailAddressError,
+		};
+	}
+
+	return { value, error };
+};
+
+const validateMailboxName = ( { value, error } ) => ( {
+	value,
+	error:
+		! error && ! /^[0-9a-z_-](\.?[0-9a-z_-])*$/i.test( value )
+			? translate( 'Only numbers, letters, dashes, underscores, and periods are allowed.' )
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Only numbers, letters, dashes, underscores, apostrophes and periods are allowed.' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/updated-email-comparison-page)